### PR TITLE
fix(calendar): refire fetch event on success after error

### PR DESCRIPTION
**Pull request by virtuacoplenny**

### CODE REVIEW

File: spot-client/src/spot-tv/calendars/calendarService.js [hunk 1]
```diff
@@ -221,8 +222,10 @@ export class CalendarService extends Emitter {
                 const events = this._updateMeetingUrlOnEvents(formattedEvents);
 
                 if (!this._hasFetchedEvents
+                    || this._hasErrorOnLastFetch
```
Comment (paweldomas) [inline spot-client/src/spot-tv/calendars/calendarService.js]: that is getting to many flags in here, I'm trying to see if there's any way to simplify things

File: spot-client/src/spot-tv/calendars/calendarService.js [hunk 2]
```diff
@@ -221,8 +222,10 @@ export class CalendarService extends Emitter {
                 const events = this._updateMeetingUrlOnEvents(formattedEvents);
 
                 if (!this._hasFetchedEvents
```
Comment (paweldomas) [inline spot-client/src/spot-tv/calendars/calendarService.js]: I don't understand why `_hasFetchedEvents` flag is needed exactly. Just don't see why comparing the last fetched events with the new ones is not enough?

File: spot-client/src/spot-tv/calendars/calendarService.js (hunk 1 above)
Comment (paweldomas) [reply spot-client/src/spot-tv/calendars/calendarService.js]: If `_calendarEvents` would be undefined initially and on error(although I'm not sure if that's even necessary) then maybe the flag could be avoided?

Comment (virtuacoplenny) [reply spot-client/src/spot-tv/calendars/calendarService.js]: You won't let me stay lazy... Good thing there are unit tests. Hopefully they're true accurate. I don't remember what edge cases were trying to be worked around with the boolean but the tests pass so maybe that's enough.
